### DISCUSSION TITLE
TransformableNode onTapListener not being called on .glb node

### DISCUSSION THREAD

**Discussion by kustraslawomir** [sceneview/sceneform-android#256] [Q&A]
Hello again.

I'm unable to recognize touch on the node (.glb). I have tried with:
```
-  TransformableNode.setOnTapListener
-  arSceneView
            .scene
            .setOnTouchListener
- hitTestResult.node (always null) etc.
```

In the video, I'm using Your example called [sample-ar-model-viewer]. Touch only works on viewRenderable, but not on .glb 3d model. I also tried to remove peek peekTouchListener after placing the object but nothing helped. 

Thank you for any tips. 

https://user-images.githubusercontent.com/16974890/146557896-58e9a6fb-00f2-424b-a4e6-00a613451fe7.mp4

**Comment by kustraslawomir** [comment]
First attempt (Toast is called only on viewRenderable tap (text/2d graphics above the character)):

```
  val transformableNode = TransformableNode(arFragment.transformationSystem).apply {
            renderable = model
            addChild(Node().apply {
                localPosition = Vector3(0.0f, 1f, 0.0f)
                localScale = Vector3(0.7f, 0.7f, 0.7f)
                renderable = modelView
            })
        }

        scene.addChild(AnchorNode(hitResult.createAnchor()).apply {
            addChild(transformableNode)
        })

        transformableNode.setOnTapListener { hitTestResult, motionEvent ->
            Toast.makeText(requireContext(), "TransformableNode tapped.", Toast.LENGTH_SHORT).show()
        }
```

**Comment by kustraslawomir** [comment]
Second attempt (hitTestResult.node is always null): 

```
        arSceneView
            .scene
            .setOnTouchListener { hitTestResult: HitTestResult, event: MotionEvent? ->
                gestureDetectorCompat.onTouchEvent(event)
                if (hitTestResult.node == null) {
                    TimberLog.w("Clicked nothing")
                    return@setOnTouchListener false
                }
                val name = hitTestResult.node?.name
                TimberLog.w("Clicked on node: $name")
                when (gestureHelper.gestureType) {
                    GestureType.NONE -> {
                    }
                    GestureType.SINGLE_TAP -> {
                    }
                    GestureType.DOUBLE_TAP -> {
                    }
                    GestureType.LONG_PRESS -> {
                    }
                }
                true
            }
```

**Comment by ThomasGorisse** [comment]
Hi,

That's because the touch event is dispatch to the ViewRenderable which could have some standard Android View clicks to handle (Button, TextView,...)

Can you please move to https://github.com/SceneView/sceneview-android in order for me not to go back inside the old Sceneform touch system code to answer your question?

We don't have yet completely renew it but setting the
https://github.com/SceneView/sceneview-android/blob/cbb9f9f6345f40b4bda18fcfa47c79fa059aba0c/sceneview/src/main/kotlin/io/github/sceneview/SceneView.kt#L407
will override any possible View touch returning true.
